avocado/utils/kernel.py

```diff
@@ -95,11 +99,26 @@ def build(self):
         Build kernel from source.
         """
         log.info("Starting build the kernel")
+        build_output_format = ""
+        if self.do_install is not None:
+            if self.distro.name == "Ubuntu":
+                build_output_format = "deb-pkg"
         if self.config_path is None:
-            build.make(self.linux_dir, extra_args='O=%s defconfig' % self.build_dir)
+            build.make(self.linux_dir, extra_args='-C %s defconfig' % self.linux_dir)
         else:
-            build.make(self.linux_dir, extra_args='O=%s olddefconfig' % self.build_dir)
-        build.make(self.linux_dir, extra_args='O=%s' % self.build_dir)
+            build.make(self.linux_dir, extra_args='-C %s olddefconfig' % self.linux_dir)
+        build.make(self.linux_dir, extra_args='-C %s %s' % (self.linux_dir, build_output_format))
+
+    def install(self):
+        """
+        Install built kernel.
+        """
+        if self.do_install is not None:
+            log.info("Starting kernel install")
+            if self.distro.name == "Ubuntu":
+                process.run('sh -c "exec sudo dpkg -i %s/*.deb"' % self.work_dir)
```

Comment:
You should be able to use `process.run()` with the `sudo` and `shell` arguments here to simplify this command.

Reply:
Did you mean "without" the `sudo` and `shell` arguments ?

If yes:
- Most systems require admin right to install a package; in our test-infrastructure, we use passwordless-sudo for automation and avoid password prompt (ei: `%sudo   ALL=(ALL:ALL) NOPASSWD: ALL`).
- The shell was invoked for two reasons:
   So that the paths to sudo and dpkg would be automatically resolved for execution.
   So that wildcart *.deb would be resolved, as the result of the kernel-build generate more than one packages to install: linux-headers, linux-libc-dev, linux-image.

Reply:
Sure, my point is that `process.run()` has parameters for both `shell` and `sudo`, so you can use something like:

```
process.run('dpkg -i %s/*.deb' % self.workdir, shell=True, sudo=True)
```

Reply:
Never knew! thank you; making the change.

Reply:
This:

http://avocado-framework.readthedocs.io/en/59.0/api/utils/avocado.utils.html

Should be handy to have in a bookmark.

Reply:
Thank you; bookmarked !